lib/src/model/comment_referable.dart

```diff
@@ -62,6 +76,50 @@ mixin CommentReferable implements Nameable {
     return result;
   }
 
+  /// Looks up references by [scope], skipping over results that do not match
+  /// the given filter.
+  ///
+  /// Override if [Scope.lookup] may return a [PrefixElement] or other elements
+  /// not corresponding to a [CommentReferable], but you still want to have
+  /// an implementation of [scope].
+  CommentReferable lookupViaScope(ReferenceChildrenLookup referenceLookup,
+      bool Function(CommentReferable) filter) {
+    var resultElement = scope.lookupPreferGetter(referenceLookup.lookup);
+    if (resultElement is PrefixElement) {
+      assert(false,
+          'PrefixElement detected, override [lookupViaScope] in subclass');
+      return null;
+    }
+    return recurseChildrenAndFilter(referenceLookup,
+        ModelElement.fromElement(resultElement, packageGraph), filter);
+  }
+
+  CommentReferable _lookupViaReferenceChildren(
+          ReferenceChildrenLookup referenceLookup,
+          bool Function(CommentReferable) filter) =>
+      recurseChildrenAndFilter(
+          referenceLookup, referenceChildren[referenceLookup.lookup], filter);
+
+  /// Assuming we found a [result], recurse through children, skipping over
```

Comment:
This comment seems to presuppose some other action. Would it make more sense for this to be private, and refer to [lookupViaScope]?

Reply:
Since implementations from other files will probably need to call it (since any of this working will depend on at least `Library` overriding `lookupViaScope` to deal with `PrefixElement`s), making it private will not work.  Made that more explicit in the comment.